### PR TITLE
[ACA-45] Drag and Drop a New Version

### PR DESCRIPTION
**Pull request by ursedaniel**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://issues.alfresco.com/jira/browse/ACA-45
Currently I can not drop a file over another file to update it's version, I can only drop a file into a folder to upload it there.

**What is the new behaviour?**

I can drag&drop a file over another file and the dialog for updating it's version will appear and after I specific the version(minor, major) and adding a comment (as optional) i can upload it's version right away.
Also I've added an option in the demo shell to enable/disable this feature.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: demo-shell/src/app/components/files/files.component.ts [hunk 1]
```diff
@@ -591,6 +596,27 @@ export class FilesComponent implements OnInit, OnChanges, OnDestroy {
         this.logService.log(event);
     }
 
+    onUploadNewVersion(ev) {
```
Comment (DenysVuika) [inline demo-shell/src/app/components/files/files.component.ts:599]: does this mean every ADF application will have to have this code implemented on the app level?

File: lib/content-services/src/lib/upload/components/upload-version-button.component.ts [hunk 2]
```diff
@@ -35,6 +35,10 @@ export class UploadVersionButtonComponent extends UploadButtonComponent implemen
     @Input()
     node: Node;
 
+    /** (**Required**) The drag&dropped new file to be versioned. */
+    @Input()
+    newFileVersion: File;
```
Comment (DenysVuika) [inline lib/content-services/src/lib/upload/components/upload-version-button.component.ts:40]: why creating second property and then assigning it to the `file`? I suggest using base.file for that

File: lib/content-services/src/lib/upload/components/upload-version-button.component.ts [hunk 3]
```diff
@@ -47,6 +51,7 @@ export class UploadVersionButtonComponent extends UploadButtonComponent implemen
     }
 
     ngOnInit() {
+        this.file = this.newFileVersion;
```
Comment (DenysVuika) [inline lib/content-services/src/lib/upload/components/upload-version-button.component.ts:54]: having 2 properties in this case does not make sense, just use `file`

File: lib/core/datatable/components/datatable/datatable.component.html [hunk 4]
```diff
@@ -60,7 +60,7 @@
                 (select)="onEnterKeyPressed(row, $event)"
                 (keyup)="onRowKeyUp(row, $event)"
                 (keydown)="onRowEnterKeyDown(row, $event)"
-                [adf-upload]="allowDropFiles && rowAllowsDrop(row)"
+                [adf-upload]="((isFolder(row) && allowDropFiles) || (!isFolder(row) && allowDropFileVersion)) && rowAllowsDrop(row)"
```
Comment (DenysVuika) [inline lib/core/datatable/components/datatable/datatable.component.html:63]: the expression is too long, can you move it to a function and use a single function call in the html? like "canUpload()"

File: lib/core/datatable/components/datatable/datatable.component.ts [hunk 5]
```diff
@@ -668,6 +679,10 @@ export class DataTableComponent implements AfterContentInit, OnChanges, DoCheck,
         return this.selectionMode && this.selectionMode.toLowerCase() === 'multiple';
     }
 
+    isFolder(row: DataRow): boolean {
```
Comment (DenysVuika) [inline lib/core/datatable/components/datatable/datatable.component.ts:682]: I think this is incorrect, the datatable component is a generic one, it should not have a notion of Folder.

File: lib/core/datatable/components/datatable/datatable.component.html (hunk 4 above)
Comment (eromano) [inline lib/core/datatable/components/datatable/datatable.component.html:63]: looks quite complicate assertion for a template maybe is better incapsulate in  a method

File: demo-shell/src/app/components/files/files.component.ts (hunk 1 above)
Comment (ursedaniel) [reply demo-shell/src/app/components/files/files.component.ts:595]: yes it does

File: lib/content-services/src/lib/upload/components/upload-version-button.component.ts (hunk 3 above)
Comment (ursedaniel) [reply lib/content-services/src/lib/upload/components/upload-version-button.component.ts]: you're right, changed that, my bad.

**Pull request merged**